lib/SILOptimizer/IPO/GlobalOpt.cpp

```diff
@@ -770,6 +786,35 @@ void SILGlobalOpt::optimizeInitializer(SILFunction *AddrF,
   HasChanged = true;
 }
 
+/// Look through all the uses of the global variable. If the only use is a
+/// store, and the global is private, then we can remove the global, the store,
+/// and the alloc. Otherwise, return.
+void SILGlobalOpt::tryOptimizeUnusedGlobal(SILGlobalVariable *global,
+                                           StoreInst *store) {
+  if (global->getLinkage() != SILLinkage::Private)
+    return;
+  if (GlobalLoadMap.count(global) || GlobalAccessMap.count(global))
```

Comment:
It might be safer to look for any use of the global except for a store, not just load and begin_access.

Also if the begin_access is for a store, you could still potentially remove the global.

Reply:
Is there a way to get all the uses of a `SILGlobalVariable`? It doesn't look like it.

Reply:
You're right, there's no way of doing that right now, so handling specific instructions that directly reference SILGlobalVariables is your best bet.